frontend/pages/admin/IntegrationsPage/cards/ConditionalAccess/ConditionalAccess.tsx

```diff
@@ -242,173 +227,210 @@ const ConditionalAccess = () => {
     },
     onError: () => {
       // distinct from successful confirmation response of `false`, this handles an API error
-      setPhase(Phase.ConfirmationError);
+      setEntraPhase(EntraPhase.ConfirmationError);
     },
   });
 
   const {
-    microsoft_entra_tenant_id: contextConfigMsetId,
-    microsoft_entra_connection_configured: contextConfigMseConfigured,
+    microsoft_entra_tenant_id: entraTenantId,
+    microsoft_entra_connection_configured: entraConfigured,
+    okta_idp_id: oktaIdpId,
   } = contextConfig?.conditional_access || {};
 
-  // only checks if tenant id already present in config, not if user added it to the form
+  // Determine if Okta is configured (all 4 fields must be present)
+  const oktaConfigured = !!(
+    oktaIdpId &&
+    contextConfig?.conditional_access?.okta_assertion_consumer_service_url &&
+    contextConfig?.conditional_access?.okta_audience_uri &&
+    contextConfig?.conditional_access?.okta_certificate
+  );
+
+  // Check if this is a managed cloud deployment (Microsoft Entra requires proxy infrastructure)
+  const isManagedCloud = contextConfig?.license?.managed_cloud || false;
+
+  // Check Entra configuration state
   useEffect(() => {
-    if (contextConfigMsetId) {
-      if (!contextConfigMseConfigured) {
-        setPhase(Phase.ConfirmingConfigured);
+    // Don't check config if we're showing the awaiting OAuth banner
+    if (showAwaitingOAuthBanner) {
+      setEntraPhase(EntraPhase.NotConfigured);
+      return;
+    }
+
+    if (entraTenantId) {
+      if (!entraConfigured) {
+        setEntraPhase(EntraPhase.ConfirmingConfigured);
       } else {
         // tenant id is present and connection is configured
-        setPhase(Phase.Configured);
+        setEntraPhase(EntraPhase.Configured);
       }
+    } else {
+      setEntraPhase(EntraPhase.NotConfigured);
     }
-  }, [contextConfigMsetId, contextConfigMseConfigured]);
+  }, [entraTenantId, entraConfigured, showAwaitingOAuthBanner]);
 
   if (!isPremiumTier) {
     return <PremiumFeatureMessage />;
   }
 
   // HANDLERS
 
-  const toggleDeleteConditionalAccessModal = () => {
-    setShowDeleteConditionalAccessModal(!showDeleteConditionalAccessModal);
+  const toggleDeleteModal = () => {
+    setShowDeleteModal(!showDeleteModal);
   };
 
-  const onSubmit = async (evt: React.FormEvent<HTMLFormElement>) => {
-    evt.preventDefault();
+  const handleEntraConnect = () => {
```

Comment:
`toggleEntraModal` would more clearly reflect what this actually does, i.e., this doesn't
actually handle connecting Entra, it just opens the modal